config/quality/checkstyle/checkstyle.xml

```diff
@@ -1,238 +1,240 @@
-            <property name="severity" value="ignore" />
+            <property name="checkFirstSentence" value="false" />
         </module>
 
+        <!-- Checks for Metrics. -->
+        <!-- See http://checkstyle.sourceforge.net/config_metrics.html -->
+        <module name="BooleanExpressionComplexity" />
+        <module name="ClassDataAbstractionCoupling" />
+        <module name="ClassFanOutComplexity" />
+        <module name="CyclomaticComplexity" />
+        <module name="JavaNCSS" />
+        <module name="NPathComplexity" />
+
+
+        <!-- Miscellaneous other checks.                   -->
+        <!-- See http://checkstyle.sourceforge.net/config_misc.html -->
+        <module name="ArrayTypeStyle" />
+        <module name="AvoidEscapedUnicodeCharacters" />
+        <module name="CommentsIndentation" />
+        <module name="DescendantToken" />
+        <module name="FileContentsHolder" />
+        <module name="FinalParameters" />
+        <module name="Indentation" />
+        <module name="OuterTypeFilename" />
+        <module name="TodoComment">
+            <property name="format" value="(?i)\s+CR\s+" />
+            <property name="severity" value="info" />
+        </module>
+        <module name="TodoComment">
+            <property name="format" value="(?i)\s+TODO\s+" />
+            <property name="severity" value="info" />
+        </module>
+        <module name="TrailingComment" />
+        <module name="UncommentedMain" />
+        <module name="UpperEll" />
+
+        <!-- Modifier Checks                                    -->
+        <!-- See http://checkstyle.sourceforge.net/config_modifiers.html -->
+        <module name="ModifierOrder" />
+        <module name="RedundantModifier" />
+
         <!-- Checks for Naming Conventions.                  -->
         <!-- See http://checkstyle.sourceforge.net/config_naming.html -->
+        <module name="AbbreviationAsWordInName" />
+        <module name="AbstractClassName" />
+        <module name="CatchParameterName" />
+        <module name="ClassTypeParameterName" />
         <module name="ConstantName" />
+        <module name="InterfaceTypeParameterName" />
         <module name="LocalFinalVariableName" />
-        <module name="FinalLocalVariable">
-            <property name="tokens" value="VARIABLE_DEF,PARAMETER_DEF"/>
-        </module>
         <module name="LocalVariableName" />
         <module name="MemberName" />
-        <module name="MethodName">
-            <property name="severity" value="ignore" />
-        </module>
+        <module name="MethodName" />
+        <module name="MethodTypeParameterName" />
         <module name="PackageName" />
-        <module name="ParameterName">
-            <property name="severity" value="ignore" />
-        </module>
-        <module name="StaticVariableName">
-            <property name="severity" value="ignore" />
-        </module>
+        <module name="ParameterName" />
```

Comment:
When I applied this rule I got exactly 0 warnings, so I think that our current codestyle already matches the default values